aws/src/main/java/org/apache/iceberg/aws/s3/S3FileIOProperties.java

```diff
@@ -428,6 +428,21 @@ public class S3FileIOProperties implements Serializable {
 
   public static final long S3_RETRY_MAX_WAIT_MS_DEFAULT = 20_000; // 20 seconds
 
+  /**
+   * Controls whether to list prefixes as directories for S3 Directory buckets Defaults value is
+   * true, where it will add the "/"
+   *
+   * <p>Example: s3://bucket/prefix will be shown as s3://bucket/prefix/
+   *
+   * <p>For more details see delimiter section in:
+   * https://docs.aws.amazon.com/AmazonS3/latest/API/API_ListObjectsV2.html#API_ListObjectsV2_RequestSyntax
+   * If set to false will throw an error when the "/" is not provided for directory bucket
```

Comment:
I think we should talk about why people would set it to false, is something like 

> Turn off this feature if you are using S3FileIO.listPrefix for listing bucket prefixes that are not directories. This would ensure correctness and fail the operation based on S3 requirement when listing against a non-directory prefix in a directory bucket.